### PR TITLE
Docs build

### PR DESCRIPTION
**Pull request by rgbkrk**
This is an attempt to get the doc builds working. Thus far I've:

* Added in the es6 container shim typings
* Adapted the after success builds to [another format](https://medium.com/@nthgergo/publishing-gh-pages-with-travis-ci-53a8270e87db) so it pushes directly to the `gh-pages` branch

However, this is still running into issues after the good old `npm install && npm run build`

```
$ npm run docs

> jupyter-services@0.0.1 docs /Users/rgbkrk/code/src/github.com/jupyter/jupyter-js-services
> typedoc --options scripts/tdoptions.json


Using TypeScript 1.5.3 from /Users/rgbkrk/code/src/github.com/jupyter/jupyter-js-services/node_modules/typescript/bin
Error: /Users/rgbkrk/code/src/github.com/jupyter/jupyter-js-services/src/kernel.ts(4)
 Cannot find module 'phosphor-signaling'.
Error: /Users/rgbkrk/code/src/github.com/jupyter/jupyter-js-services/src/kernel.ts(163)
 Type 'any' is not generic.
Error: /Users/rgbkrk/code/src/github.com/jupyter/jupyter-js-services/src/session.ts(4)
 Cannot find module 'phosphor-signaling'.
Error: /Users/rgbkrk/code/src/github.com/jupyter/jupyter-js-services/src/session.ts(59)
 Type 'any' is not generic.

npm ERR! Darwin 14.5.0
npm ERR! argv "/Users/rgbkrk/.nvm/versions/io.js/v3.0.0/bin/iojs" "/Users/rgbkrk/.nvm/versions/io.js/v3.0.0/bin/npm" "run" "docs"
npm ERR! node v3.0.0
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE
npm ERR! jupyter-services@0.0.1 docs: `typedoc --options scripts/tdoptions.json`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the jupyter-services@0.0.1 docs script 'typedoc --options scripts/tdoptions.json'.
npm ERR! This is most likely a problem with the jupyter-services package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     typedoc --options scripts/tdoptions.json
npm ERR! You can get their info via:
npm ERR!     npm owner ls jupyter-services
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/rgbkrk/code/src/github.com/jupyter/jupyter-js-services/npm-debug.log
```

For some reason it's not picking up the phosphor signaling module when using typedoc.

Not sure what the next step is here.

/cc @sccolbert @blink1073

### CODE REVIEW

**Comment by blink1073** [comment]
Whoops:

```diff
diff --git a/scripts/tdoptions.json b/scripts/tdoptions.json
index a94f534..707a59d 100644
--- a/scripts/tdoptions.json
+++ b/scripts/tdoptions.json
@@ -10,6 +10,7 @@
     "typings/es6-promise/es6-promise.d.ts",
     "typings/text-encoding/text-encoding.d.ts",
     "typings/es6-container-shim.d.ts",
-    "typings/es6-dataview.d.ts"
+    "typings/es6-dataview.d.ts",
+    "node_modules/phosphor-signaling/lib/phosphor-signaling.d.ts"
   ]
 }
```

**Comment by rgbkrk** [comment]
Ah!

**Comment by rgbkrk** [comment]
:boom:

```
$ git apply ./patch
15:33:24 {doccie *%=} kyle6475@lack ~/code/src/github.com/jupyter/jupyter-js-services$ npm run docs

> jupyter-services@0.0.1 docs /Users/kyle6475/code/src/github.com/jupyter/jupyter-js-services
> typedoc --options scripts/tdoptions.json


Using TypeScript 1.5.3 from /Users/kyle6475/code/src/github.com/jupyter/jupyter-js-services/node_modules/typescript/bin
Rendering [========================================] 100%

Documentation generated at /Users/kyle6475/code/src/github.com/jupyter/jupyter-js-services/docs
```

**Comment by rgbkrk** [comment]
All set now!

**Comment by blink1073** [comment]
The page automatically redirects to `jupyter.org`, should that be the `CNAME`?

**Comment by rgbkrk** [comment]
Ah, I should take the CNAME comment out. This will appear at jupyter.org/jupyter-js-services since jupyter.github.io already points to jupyter.org via CNAME.

**Comment by blink1073** [comment]
Boom.

**Comment by blink1073** [comment]
Uh oh: https://travis-ci.org/jupyter/jupyter-js-services#L300

**Comment by rgbkrk** [comment]
Yeah.

I had `GH_REF` set then deleted it right after.

```diff
diff --git a/scripts/travis_after_success.sh b/scripts/travis_after_success.sh
index 26acf64..3041603 100644
--- a/scripts/travis_after_success.sh
+++ b/scripts/travis_after_success.sh
@@ -2,6 +2,8 @@
 if [[ $TRAVIS_PULL_REQUEST == false && $TRAVIS_BRANCH == "master" ]]
 then
     echo "-- building docs --"
+    GH_REF="https://github.com/jupyter/jupyter-js-services.git"
+
     npm run docs

     ( cd docs
```

**Comment by rgbkrk** [comment]
(PR incoming)

File: scripts/travis_after_success.sh [hunk 1]
```diff
@@ -1,26 +1,18 @@
...
-    git config credential.helper "store --file=.git/credentials"
-
-    rm -rf ./*
-    cp -r ../build/docs/* ./.
-    git add -A
-    git commit -m "autocommit docs"
-    git push origin gh-pages
+    git add .
+    git commit -m "Deployed to GitHub Pages"
+    git push --force --quiet "https://${GHTOKEN}@${GH_REF}" master:gh-pages /dev/null 2>&1
```
Comment (blink1073) [inline scripts/travis_after_success.sh]: We don't have a GH_REF defined in our .travis.yml

**Pull request merged**